### PR TITLE
Option for giving snapshots as source disks for instances

### PR DESCRIPTION
**Pull request by tahir24434**
Signed-off-by: tahir24434 tahir.rauf1@gmail.com

### CODE REVIEW

File: src/main/python/hydra_deploy/setup_config.ini [hunk 1]
```diff
@@ -10,8 +10,12 @@ zone: <Zone where you want to spawn instances>
 EmailId: <Your email_id>
 Disk1Type: pd-standard
 Disk2Type: pd-standard
+; Options are ubuntu-12-04 or ubuntu-14-04. You can also mention the source snapshots of disk1 and disk2 as well.
+Disk1Image: plumgrid-vm-builder-v7-d1-1464853307
```
Comment (sushilks) [inline src/main/python/hydra_deploy/setup_config.ini]: Best to leave exact image name out of this. <Your imagename is sufficient>

File: src/main/python/hydra_deploy/hydra_setup_script.py [hunk 2]
```diff
@@ -12,12 +12,12 @@ def setup(step):
         setup_helpers.run_cmd_on_host(dst_user_name, mesos_masters_ips_list[0], "sudo apt-get -y install git unzip")
         setup_helpers.run_cmd_on_host(dst_user_name, mesos_masters_ips_list[0],
                                       "wget https://github.com/sushilks/hydra/archive/master.zip && unzip master.zip")
-        print("==> Install protobuf on all nodes")
-        setup_helpers.run_cmd_on_multiple_hosts(
-            dst_user_name, mesos_all_ips_list,
-            "wget http://launchpadlibrarian.net/160197953/libprotobuf7_2.4.1-3ubuntu4_amd64.deb")
-        setup_helpers.run_cmd_on_multiple_hosts(dst_user_name, mesos_all_ips_list,
-                                                "dpkg -i ./libprotobuf7_2.4.1-3ubuntu4_amd64.deb", use_sudo=True)
+        # print("==> Install protobuf on all nodes")
```
Comment (sushilks) [inline src/main/python/hydra_deploy/hydra_setup_script.py]: Can you add shebang

File: src/main/python/hydra_deploy/mesos_marathon_setup.py [hunk 3]
```diff
@@ -194,8 +194,9 @@ def setup(step, config):
     parser.add_argument('--config_file', '-f', type=str,
                         default=os.getcwd() + "/setup_config.ini",
                         help='Setup configuration file. File holds your gcloud confiuration settings. It also holds '
-                             'values of number of master/slave nodes.A sample configuration file can be found at '
-                             'hydra-deploy/src/main/python/gcloud/setup_config.ini. ')
+                             'values of number of master/slave nodes. A sample configuration file can be found at '
+                             'hydra-deploy/src/main/python/gcloud/setup_config.ini (This is also default conf file). ')
+
```
Comment (sushilks) [inline src/main/python/hydra_deploy/mesos_marathon_setup.py:204]: The master script currently just exists at the end of the install, 
it should have a nice success message with the list of instances installed and there ip addresses.

File: src/main/python/hydra_deploy/mesos_marathon_setup.py [hunk 4]
```diff
@@ -22,17 +22,17 @@ def setup(step, config):
                 tag = options_dict["tag"]
                 for i in range(int(count)):
                     instance_name = deployment_id + "-" + section + "-" + tag + "-" + str(i)
-                    setup_helpers.spawn_instance(instance_name, "ubuntu-14-04", dst_user_name, ssh_key_file, config,
+                    setup_helpers.spawn_instance(instance_name, dst_user_name, ssh_key_file, config,
```
Comment (sushilks) [inline src/main/python/hydra_deploy/mesos_marathon_setup.py]: please come up with a smaller name for the launch utility.

File: src/main/python/hydra_deploy/setup_config.ini (hunk 1 above)
Comment (tahir24434) [inline src/main/python/hydra_deploy/setup_config.ini]: Done.

File: src/main/python/hydra_deploy/hydra_setup_script.py (hunk 2 above)
Comment (tahir24434) [inline src/main/python/hydra_deploy/hydra_setup_script.py]: Done.

File: src/main/python/hydra_deploy/mesos_marathon_setup.py (hunk 3 above)
Comment (tahir24434) [inline src/main/python/hydra_deploy/mesos_marathon_setup.py:204]: Done.

**Pull request merged**